charts/opentelemetry-collector/UPGRADING.md

```diff
@@ -1,5 +1,32 @@
 # Upgrade guidelines
 
+## 0.23.0 to 0.24.0
+
+[Remove containerLogs in favor of presets.logsCollection]()
+
+The ability to enable logs collection from the collector has been moved from `containerLogs.enabled` to `presets.logsCollection.enabled`. If you are currently using `containerLogs.enabled`, you should instead use the preset:
+
+```yaml
+presets:
+  logsCollection:
+    enabled: true
+```
+
+If you are using `containerLogs.enabled` and also enabling collection of the collector logs you can use `includeCollectorLogs`
+
+```yaml
+presets:
+  logsCollection:
+    enabled: true
+    includeCollectorLogs: true
```

Comment:
What if we add `filelogConfig` here?

```
presets:
  logsCollection:
    enabled: true
    includeCollectorLogs: true
    filelogConfig: 
       ...
```

Same for hostmetrics preset. Then it becomes fully overridable. `exclude` can be adjusted with `set` from http://masterminds.github.io/sprig/dicts.html if includeCollectorLogs=false.

Reply:
I thought about that too.  Feels natural for hostmetrics, but the default filelog config is massive and it felt strange to add it to the values.yaml.  The benefits probably outweigh the messy values.yaml tho.  I'll try it out this week.

Reply:
I think this is the only sticking point:

`exclude: [ /var/log/pods/{{ .Release.Namespace }}_{{ include "opentelemetry-collector.fullname" . }}*_*/{{ .Chart.Name }}/*.log ]
`

The filelog config today is built with some knowledge about where the collector is going to run.  If we switch the default for `includeCollectorLogs` to be `true` we can default the config to be `exclude: []` in values.yaml and then then we can use the same `append`/`set` logic to update the exclude list to the value above if `includeCollectorLogs` is switched to `false`.

Reply:
Even if `includeCollectorLogs: false` and `exclude: []` by default, we can do the same `_ := set exclude ... append | uniq` trick in the templating

Reply:
> Feels natural for hostmetrics, but the default filelog config is massive and it felt strange to add it to the values.yaml

You're right. I just checked the filelog config. It's too big, and overriding operators would be not convenient at all. My only concern is making `scrapers` configurable with enabled hostmetrics preset. There should be another way to do that

Reply:
Maybe it's as simple as adding the hostmetrics config to the interface but not filelog? Or we reduce the number of default scrapers and make it less likely a user would need to remove something from the default config.

Reply:
But if we do add hostmetrics config to the interface then users would be able to add/update/remove anything as if it was a default entry in .Values.config.

Reply:
> Or we reduce the number of default scrapers and make it less likely a user would need to remove something from the default config.

Let's probably do this. It's always possible to disable particular metrics if needed

Reply:
Better to start with smaller configuration interface for presets and enhance it if needed in future

Reply:
@dmitryax done.  Reduced to cpu, load, memory, disk, and network.